Userland/Libraries/LibJS/Runtime/Intl/NumberFormatPrototype.cpp

```diff
@@ -56,6 +58,23 @@ JS_DEFINE_NATIVE_FUNCTION(NumberFormatPrototype::format)
     return number_format->bound_format();
 }
 
+// 15.4.4 Intl.NumberFormat.prototype.formatToParts ( value ), https://tc39.es/ecma402/#sec-intl.numberformat.prototype.formattoparts
+JS_DEFINE_NATIVE_FUNCTION(NumberFormatPrototype::format_to_parts)
+{
+    auto value = vm.argument(0);
+
+    // 1. Let nf be the this value.
+    // 2. Perform ? RequireInternalSlot(nf, [[InitializedNumberFormat]]).
+    auto* number_format = TRY(typed_this_object(global_object));
+
+    // 3. Let x be ? ToNumeric(value).
+    value = TRY(value.to_numeric(global_object));
+
+    // 4. Return ? FormatNumericToParts(nf, x).
+    // Note: Our implementation of FormatNumericToParts does not throw.
+    return format_numeric_to_parts(global_object, *number_format, value.as_double());
```

Comment:
This crashes when passing a bigint to the function

Reply:
Hmm, then `Intl.NumberFormat.prototype.format` has the same issue. Looking into it.

Reply:
Is `value.as_bigint().big_integer().to_double()` a normal thing to do? Does that lose any precision?

Reply:
That works if a certain bigint is known to be small enough, and we need to do some floating-point math with it (for example). Since the given argument can be arbitrarily large, it might not end well :^)

```
double UnsignedBigInteger::to_double() const
{
    // FIXME: I am naive
    return static_cast<double>(to_u64());
}
```

I suggest we just check for `is_bigint()` after the `to_number()` call and throw a `NotImplemented` TypeError. Then we can add a variant of `format_numeric_to_parts` that can deal with bigints later.

Reply:
:+1: sounds good. I added a commit to do the same for `Intl.NumberFormat.prototype.format` as well.